[SPARK-9649] Fix flaky test MasterSuite again - disable REST

andrewor14 opened the pull request:
The REST server is not actually used in most tests and so we can disable it. It is a source of flakiness because it tries to bind to a specific port in vain. There was also some code that avoided the shuffle service in tests. This is actually not necessary because the shuffle service is already off by default.

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

JoshRosen commented:
Are there any tests where we need to explicitly re-enable this configuration or is this feature completely untested?

SparkQA commented:
[Test build #40343 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40343/consoleFull) for   PR 8084 at commit [`f86f2be`](https://github.com/apache/spark/commit/f86f2be5c637549de72bb6d7aba46a513a645d06).

andrewor14 commented:
It is tested in `RestSubmissionSuite` but is mocked out there and doesn't go through the standalone `Master.

SparkQA commented:
**[Test build #40343 timed out](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40343/console)**     for PR 8084 at commit [`f86f2be`](https://github.com/apache/spark/commit/f86f2be5c637549de72bb6d7aba46a513a645d06)     after a configured wait of `175m`.

AmplabJenkins commented:
Merged build finished. Test FAILed.

JoshRosen commented:
Jenkins, retest this please.

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #40411 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40411/consoleFull) for   PR 8084 at commit [`f86f2be`](https://github.com/apache/spark/commit/f86f2be5c637549de72bb6d7aba46a513a645d06).

SparkQA commented:
**[Test build #40411 timed out](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40411/console)**     for PR 8084 at commit [`f86f2be`](https://github.com/apache/spark/commit/f86f2be5c637549de72bb6d7aba46a513a645d06)     after a configured wait of `175m`.

AmplabJenkins commented:
Merged build finished. Test FAILed.

andrewor14 commented:
retest this please

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

AmplabJenkins commented:
Merged build finished. Test FAILed.

andrewor14 commented:
retest this please?

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #40475 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40475/consoleFull) for   PR 8084 at commit [`f86f2be`](https://github.com/apache/spark/commit/f86f2be5c637549de72bb6d7aba46a513a645d06).

SparkQA commented:
**[Test build #40475 timed out](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40475/console)**     for PR 8084 at commit [`f86f2be`](https://github.com/apache/spark/commit/f86f2be5c637549de72bb6d7aba46a513a645d06)     after a configured wait of `175m`.

AmplabJenkins commented:
Merged build finished. Test FAILed.

Commit:
Do not enable REST server in tests

There was some code that also avoided the shuffle service in tests.
This is actually not necessary because the shuffle service is
already off by default.

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #1456 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1456/consoleFull) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).

AmplabJenkins commented:
Merged build finished. Test FAILed.

andrewor14 commented:
retest this please

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #40523 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40523/consoleFull) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).

SparkQA commented:
[Test build #1456 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1456/console) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).
 * This patch **fails Spark unit tests**.
 * This patch **does not merge cleanly**.
 * This patch adds no public classes.

andrewor14 commented:
retest this please

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #1466 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1466/consoleFull) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).

SparkQA commented:
[Test build #1465 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1465/consoleFull) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).

SparkQA commented:
**[Test build #40523 timed out](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40523/console)**     for PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f)     after a configured wait of `175m`.

AmplabJenkins commented:
Merged build finished. Test FAILed.

JoshRosen commented:
Jenkins, retest this please.

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #40575 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40575/consoleFull) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).

AmplabJenkins commented:
Merged build finished. Test FAILed.

SparkQA commented:
[Test build #1465 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1465/console) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

JoshRosen commented:
Jenkins, retest this please.

JoshRosen commented:
Wow, this has been the unluckiest PR when it comes to flakiness :(

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #40580 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40580/consoleFull) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).

SparkQA commented:
[Test build #1466 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/1466/console) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

JoshRosen commented:
LGTM, so I'm going to merge this to master and branch-1.5.

SparkQA commented:
[Test build #40575 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40575/console) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

SparkQA commented:
[Test build #40580 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/40580/console) for   PR 8084 at commit [`026f628`](https://github.com/apache/spark/commit/026f6287086be12c7bbbb0633f38e333d26d831f).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

JoshRosen commented:
@andrewor14 do you think that this is also a candidate for backport to 1.4.x? I think that I've seen similar MasterSuite flakiness there as well.

andrewor14 commented:
Yeah, I think it doesn't hurt.
